XRTK-Core/Packages/com.xrtk.core/Services/MixedRealityToolkit.cs

```diff
@@ -753,12 +753,12 @@ private void OnApplicationPause(bool pause)
         #region Registration
 
         /// <summary>
-        /// Registers all the services defined in the provided configuration collection.
+        /// Registers all the <see cref="IMixedRealityService"/>s defined in the provided configuration collection.
         /// </summary>
-        /// <typeparam name="T"></typeparam>
-        /// <param name="configurations">The list of <see cref="MixedRealityServiceConfiguration"/>s</param>
+        /// <typeparam name="T">The interface type for the <see cref="IMixedRealityService"/> to be registered.</typeparam>
+        /// <param name="configurations">The list of <see cref="MixedRealityServiceConfiguration{T}"/>s.</param>
         /// <returns>True, if all configurations successfully created and registered their services.</returns>
-        public static bool RegisterServices<T>(MixedRealityServiceConfiguration<T>[] configurations) where T : IMixedRealityService
+        public static bool TryRegisterServiceConfigurations<T>(MixedRealityServiceConfiguration<T>[] configurations) where T : IMixedRealityService
```

Comment:
I can't see any use of this function in this change, is this needed?

The description also does not reflect what this function does.

Reading:
* It seems to take a collection of service configurations
* Register the service found in each service configuration
* Register the configuration for any data providers for any services registered

If it's needed fine but it doesn't read right?

Reply:
It's not currently used, but could be by someone using the toolkit. If we can register data providers this way it's only reasonable that we can do the same for services as well.

I'm confused about it not reading right. What you write above is correct, and that's what is reflected in the method.

Reply:
Added the usage of this method for registering the extension service configurations in `InitializeServiceLocator()`